File: branches/scala/api/src/main/scala/org/vosao/global/FileCache.scala
package org.vosao.global

import org.vosao.entity.FileEntity

trait FileCache {

	def isInCache(path: String): Boolean
	
	def isInPublicCache(path: String): Boolean
	
	def put(path: String, file: FileEntity, content: Array[Byte]): Unit
	
	def getFile(path: String): FileEntity
	
	def getContent(path: String): Array[Byte]
	
	def makePublic(path: String): Unit

	def remove(path: String): Unit
	
}

File: tags/0.5/api/src/main/java/org/vosao/global/FileCache.java
package org.vosao.global;

import org.vosao.entity.FileEntity;

public interface FileCache {

	boolean isInCache(String path);
	
	boolean isInPublicCache(String path);
	
	void put(String path, FileEntity file, byte[] content);
	
	FileEntity getFile(String path);
	
	byte[] getContent(String path);
	
	void makePublic(String path);

	void remove(String path);
	
}
